src/coreclr/jit/lower.cpp

```diff
@@ -8179,8 +8185,354 @@ void Lowering::LowerIndir(GenTreeIndir* ind)
         const bool isContainable = false;
         TryCreateAddrMode(ind->Addr(), isContainable, ind);
     }
+
+#ifdef TARGET_ARM64
+    if (comp->opts.OptimizationEnabled() && ind->OperIs(GT_IND))
+    {
+        OptimizeForLdp(ind);
+    }
+#endif
+
+    return next;
 }
 
+#ifdef TARGET_ARM64
+
+//------------------------------------------------------------------------
+// OptimizeForLdp: Record information about an indirection, and try to optimize
+// it by moving it to be adjacent with a previous indirection such that they
+// can be transformed into 'ldp'.
+//
+// Arguments:
+//    ind - Indirection to record and to try to move.
+//
+// Returns:
+//    True if the optimization was successful.
+//
+bool Lowering::OptimizeForLdp(GenTreeIndir* ind)
+{
+    if (!ind->TypeIs(TYP_INT, TYP_LONG, TYP_DOUBLE, TYP_SIMD16))
+    {
+        return false;
+    }
+
+    target_ssize_t offs   = 0;
+    GenTree*       addr   = ind->Addr();
+    FieldSeq*      fldSeq = nullptr;
+    comp->gtPeelOffsets(&addr, &offs, &fldSeq);
```

Comment:
`fldSeq` is unused here; don't compute it?

Reply:
Made it optional.